conda_build/index.py

```diff
@@ -289,6 +289,9 @@ def _read_index_tar(tar_path, lock, locking=True, timeout=90):
         with tarfile.open(tar_path) as t:
             try:
                 index_json = json.loads(t.extractfile('info/index.json').read().decode('utf-8'))
+                # add unsatisfiable dependency to revoked packages
+                if isfile(tar_path + '.REVOKED'):
+                    index_json['depends'].append('package_has_been_revoked')
```

Comment:
I can cave on this solution for backward compatibility, but I don't want it as a requirement moving forward.  Can we also add:

    index_json['revoked'] = True

?